data-model/src/main/java/io/micronaut/data/model/query/builder/AbstractSqlLikeQueryBuilder.java

```diff
@@ -1437,8 +1438,18 @@ private void appendCriteriaForOperator(StringBuilder whereClause,
             }
 
         } else {
+            if (isInOperator && propertyPath.getProperty() instanceof Embedded) {
```

Comment:
This was not supported before and it's pretty complex to support now, so for the time being we can throw exception when query model with actual embedded collection params is passed for IN clause. Probably not used too often, so we can focus on it if needed later.